lib/new_relic/agent/instrumentation/rack.rb

```diff
@@ -99,27 +99,31 @@ def run_with_newrelic(app, *args, **kwargs)
             else
               run_without_newrelic(app, *args, **kwargs)
             end
-          end          
+          end
         end
 
         if RUBY_VERSION < "2.7.0"
           def use_with_newrelic(middleware_class, *args, &blk)
-            if ::NewRelic::Agent::Instrumentation::RackHelpers.middleware_instrumentation_enabled?
-              wrapped_middleware_class = ::NewRelic::Agent::Instrumentation::MiddlewareProxy.for_class(middleware_class)
-              use_without_newrelic(wrapped_middleware_class, *args, &blk)
-            else
-              use_without_newrelic(middleware_class, *args, &blk)
+            unless middleware_class.nil?
```

Comment:
use guard clause instead of adding another conditional block around the method's implementation.

Reply:
That is a bit too much nesting, isn't it? I will happily make that change for readability.